### DISCUSSION TITLE
Versioning policy

### DISCUSSION THREAD

**Discussion by davidhunter22** [microsoft/vcpkg#19244] [Q&A]
Is there a versioning policy for VCPKG?

I notice that there are tags of the 2020.04 form but they seem to occur fairly randomly in that some months have one and some don't and when they appear during or sometimes after the month indicated by the tag is unclear.

So in general is there any policy around creating tags? There was an issue https://github.com/microsoft/vcpkg/issues/10569 that talked about a specific tag. Some comments were of the "live at head", "use a git revision". However in a larger setting I find those problematic. If you have a large set of users you can't say "live at head" because that may change every minute. Even live at git revision is difficult as many users can't even modify the central vcpkg they are using, also telling people  use "6185aa76504a5025f36754324abf307cc776f3da" makes many think your insane!!! Version number of the form 2020.04 convey a large amount of information an implictly indicate all the version numbers of libraries you are using. 

Now maybe this will be fixed by manifest files, local per project builds, binary caching etc... all of which are awesome and I can't wait to use them. On a side note on the roadmap  https://github.com/microsoft/vcpkg/wiki/Roadmap all those big features have realease date versions in the past, so yeah :-). Anyway in the mean time it would be nice to know what a users expectation should be around tagged version.

Maybe there is a distiction here between the version of vcpkg itself and the libraries you can use via vcpkg. To me this would imply splitting the project into a "core" and libraries that use that core but I'm probably wrong ( as usual! ).

On a related topic it will be interesting to see what happens with version once vcpkg comes along with Visual Studio [Code] as presumably these won't change during a given version of VS like 16.7.

**Comment by strega-nil** [comment]
This is definitely a thing we've been discussing; we don't have an answer yet. Hopefully once we have shipping binaries, we can discuss this further, but until that point it's likely that we won't have a good versioning story. That means probably by end-of-year. Unfortunately, this requires quite a lot of effort and time; we are currently working on it, however.

**Comment by davidhunter22** [comment]
Is it possible to at least carry on doing the monthly tags whenver you think the project is in a reasonable shape, with no gaurantees, in the mean time?

**Comment by davidhunter22** [comment]
Note that I feel that versioning is more of an issue for vcpkg in it's current form than say NuGet. I generally don't care much what version of NuGet I am running as the versions of the libraries I use via NuGet are independent. If vcpkg evolves into this model eventually then I probably won't care so much about the version of vcpkg.exe that I am running either. In other words the tight coupling of the version of vcpkg I am using, by tag of git version, with the versions of the libraries I use via it is the root of the problem.

**Comment by strega-nil** [comment]
Quite. I'll see what I can do.

**Comment by BillyONeal** [comment]
Ping @strega-nil we didn't know why this was requires:discussion today

**Comment by strega-nil** [comment]
@BillyONeal because we need to talk about versioning. This is less a "requires:discussion" and more of a "requires:deep-technical-discussion" haha

**Comment by BillyONeal** [comment]
OK, I'm dropping the tag (since that's the 'discuss in the Thursday meeting' tag) and assigning to Victor since it's his area. @vicroms we'll be looking on you to "drive" here.

**Comment by PhoebeHui** [comment]
@davidhunter22, does this still be a question for you? the vcpkg-tool has its own release process https://github.com/microsoft/vcpkg-tool/releases. For the libraries, the release information in https://github.com/microsoft/vcpkg/releases
